README.md

```diff
@@ -17,26 +23,191 @@ The SDK handles the device data collection, communication with the card issuer,
+
+```swift
+   private var transaction: Transaction?
+   private var standalone3DSService: ThreeDS2Service?
+````
+
+2. Initialize the SDK with your preferred user interface options:
+
+- `scheme` can be set only as `visa` and `mastercard` in lower case string.
+
+```swift
+// initialise Standalone 3DS Service with required parameters
+
+do {
+   let directoryServerData = ThreeDS2ServiceConfiguration.DirectoryServerData(directoryServerID: <directoryServerID>,
+                                                                              directoryServerPublicKey: <ds_public>,
+                                                                              directoryServerRootCertificates: [<caPublic>])
+   let configParameters = ThreeDS2ServiceConfiguration.ConfigParameters(directoryServerData: directoryServerData,
+                                                                        messageVersion: <messageVersion>,
+                                                                        scheme: <scheme>)
+    let serviceConfiguration = ThreeDS2ServiceConfiguration(configParameters: configParameters)
+      
+    self.standalone3DSService = try Standalone3DSService.initialize(with: serviceConfiguration)
+   }  catch let error {
+      // handle failure scenario
+   }
+```
+
+3. Create `transaction` using created `ThreeDS2Service` object:
+
+```swift
+  self.transaction = self.standalone3DSService?.createTransaction()
+```
+
+4. Retrieve the `authenticationRequestParameters`:
+```swift
+ // get Authentication Request parameters
+ self.transaction?.getAuthenticationRequestParameters { result in
+     switch result {
+     case .success(let params):
+        //Make an Authentication Request to your 3DS Server
+     case .failure(let error):
+        // handle failure scenario
+      }
+}
+
+```
+5. Handle the `doChallenge` flow: 
+
+- If the Authentication response that is returned indicates that the Challenge Flow must be applied, the 3DS Requestor App calls the `doChallenge` method with the required input `ChallengeParameters`. The `doChallenge` method initiates the challenge process.
+- The `doChallenge` function now returns a `ChallengeResult` object with `transactionStatus` and `sdkTransactionID`.
+```swift
+let params = ChallengeParameters(threeDSServerTransactionID: response.transactionId,
+                                 acsTransactionID: response.acs.transactionId,
+                                 acsRefNumber: response.acs.referenceNumber,
+                                 acsSignedContent: response.acs.signedContent)
+transaction?.doChallenge(challengeParameters: params, completion: { [weak self]  result in
+     switch result {
+      case .success(let result):
+          // handle success scenario
+      case .failure(let error):
+         // handle failure scenario
+       }
+      // call close and cleanUp methods after challenge flow is completed. 
+      self?.transaction?.close()
+      self?.standalone3DSService?.cleanUp()
```

Comment:
re, I'm not aware of the actual reason so a bit of an uneducated guess here: Why don't we have a strong reference to `self` in this closure to avoid, for example, `close()` to be called but `self` being deallocated and `cleanUp()` is not being called?